File: medplum-java/medplum-model/src/main/java/com/fhir/Bundle.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableBundle.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface Bundle {
  Optional<code> language();

  Optional<List<Bundle_Entry>> entry();

  Optional<uri> implicitRules();

  Optional<id> id();

  Optional<Meta> meta();

  Optional<Signature> signature();

  Optional<Identifier> identifier();

  Optional<unsignedInt> total();

  String resourceType();

  Optional<List<Bundle_Link>> link();

  Optional<instant> timestamp();

  Optional<BundleType> type();

  static ImmutableBundle.ResourceTypeBuildStage builder() {
    return ImmutableBundle.builder();
  }
}

File: medplum-typescript/src/model/fhir/Bundle.ts
import { BundleType } from "./BundleType";
import { Bundle_Entry } from "./Bundle_Entry";
import { Bundle_Link } from "./Bundle_Link";
import { code } from "./code";
import { id } from "./id";
import { Identifier } from "./Identifier";
import { instant } from "./instant";
import { Meta } from "./Meta";
import { Signature } from "./Signature";
import { unsignedInt } from "./unsignedInt";
import { uri } from "./uri";

export interface Bundle {
    resourceType: string;
    id: id | null | undefined;
    meta: Meta | null | undefined;
    implicitRules: uri | null | undefined;
    language: code | null | undefined;
    identifier: Identifier | null | undefined;
    type: BundleType | null | undefined;
    timestamp: instant | null | undefined;
    total: unsignedInt | null | undefined;
    link: Bundle_Link[] | null | undefined;
    entry: Bundle_Entry[] | null | undefined;
    signature: Signature | null | undefined;
}
